WebHostLib/downloads.py

```diff
@@ -72,7 +72,10 @@ def download_slot_file(room_id, player_id: int):
                     if name.endswith("info.json"):
                         fname = name.rsplit("/", 1)[0] + ".zip"
         elif slot_data.game == "Ocarina of Time":
-            fname = f"AP_{app.jinja_env.filters['suuid'](room_id)}_P{slot_data.player_id}_{slot_data.player_name}.apz5"
+            with zipfile.ZipFile(io.BytesIO(slot_data.data)) as zf:
```

Comment:
This looks like it will crash with a non-descript error on trying to upload an older patch?

Reply:
zipfile.is_zipfile might help, assuming the old format is not a valid zip file

Reply:
They are not, so I will make that change.